crates/extensions/c8y_mapper_ext/src/inventory.rs

```diff
@@ -4,58 +4,23 @@ use crate::error::ConversionError;
 use crate::fragments::C8yAgentFragment;
 use serde_json::json;
 use serde_json::Value as JsonValue;
-use std::fs::File;
-use std::io::Read;
-use std::path::Path;
 use tedge_api::entity::EntityType;
 use tedge_api::entity_store::EntityTwinMessage;
-use tedge_api::mqtt_topics::Channel;
 use tedge_api::mqtt_topics::EntityTopicId;
 use tedge_config::models::TopicPrefix;
 use tedge_mqtt_ext::MqttMessage;
 use tedge_mqtt_ext::Topic;
-use tracing::info;
 
-const INVENTORY_FRAGMENTS_FILE_LOCATION: &str = "device/inventory.json";
 const INVENTORY_MANAGED_OBJECTS_TOPIC: &str = "inventory/managedObjects/update";
 
 impl CumulocityConverter {
     /// Creates the inventory twin messages with fragments from inventory.json file
     pub(crate) fn base_inventory_twin_data(&mut self) -> Result<Vec<MqttMessage>, ConversionError> {
-        let mut messages = vec![];
-        let inventory_file_path = self
-            .config
-            .config_dir
-            .join(INVENTORY_FRAGMENTS_FILE_LOCATION);
-        let inventory_base = Self::get_inventory_fragments(inventory_file_path.as_std_path())?;
-
-        if let JsonValue::Object(mut map) = inventory_base {
-            map.entry("name").or_insert(json!(self.device_name));
-            map.entry("type").or_insert(json!(self.device_type));
```

Comment:
The mapper publishing the device name and type from the tedge config as twin data on startup was already wrong, as this would have lead to one mapper overwriting the name overridden by another mapper in the config (since different cloud mappers can have their own device ids). The agent shouldn't be doing this either, as the impact would be the same and hence this logic was simply dropped and not ported to the agent either.